### ISSUE TITLE
Wrong transpiled value of enums with string values

### ISSUE THREAD

**Issue by felixmosh** [swc-project/swc#3361]
### Describe the bug

Enums with string values are generated from some reason as the enum name instead of using the value.

### Input code

```typescript
export enum CallType {
  'ALL' = 'all',
  'MY_CALLS' = 'my-calls',
  'NOT_ANSWERED' = 'not-answered',
}
```

### Playground link

https://play.swc.rs/?version=1.2.133&code=H4sIAAAAAAAAA0utKMgvKlFIzSvNVXBOzMkJqSxIVajmUlBQd%2FTxUVewVVAHCqrrgAR8I%2BOdgYLBYNHcSt1koEwxRMrPPyTe0S843DXI1QUsnZdfopuYV1yeWpSaAlRSywUA47fZDWgAAAA%3D&config=H4sIAAAAAAAAA0WNSwrDMAxE76J1ts0id%2BghjKsEF%2F%2FQKBBjfPfawSU7aebpqdIXlrZK2QhYxoQS1Vy0kZbMsOKy0kKKHu3Gg1tfjBysHWG8eudTAs92oeCi28sw2RSyMPBUJh7%2BT7YuCulzjqDez27hSu1xzDuH9wRVTm4%2FOkwQUrUAAAA%3D

### Expected behavior

It should use the given values, like TSC does
```js
export var CallType;
(function (CallType) {
    CallType["ALL"] = "all";
    CallType["MY_CALLS"] = "my-calls";
    CallType["NOT_ANSWERED"] = "not-answered";
})(CallType || (CallType = {}));
```

### Actual behavior

export var CallType;
(function(CallType) {
    CallType['ALL'] = 'ALL';
    CallType['MY_CALLS'] = 'MY_CALLS';
    CallType['NOT_ANSWERED'] = 'NOT_ANSWERED';
})(CallType || (CallType = {}));

### Version

1.2.133

### Additional context

I've tested it and seems that this is a regression bug started at `v1.2.130`.
It was fixed in `v1.2.134` which is **not published** yet

**Comment by magic-akari** [comment]
fixed in [1.2.134](https://play.swc.rs/?version=1.2.134&code=H4sIAAAAAAAAA0utKMgvKlFIzSvNVXBOzMkJqSxIVajmUlBQd%2FTxUVewVVAHCqrrgAR8I%2BOdgYLBYNHcSt1koEwxRMrPPyTe0S843DXI1QUsnZdfopuYV1yeWpSaAlRSywUA47fZDWgAAAA%3D&config=H4sIAAAAAAAAA0WNSwrDMAxE76J1ts0id%2BghjKsEF%2F%2FQKBBjfPfawSU7aebpqdIXlrZK2QhYxoQS1Vy0kZbMsOKy0kKKHu3Gg1tfjBysHWG8eudTAs92oeCi28sw2RSyMPBUJh7%2BT7YuCulzjqDez27hSu1xzDuH9wRVTm4%2FOkwQUrUAAAA%3D)

**Comment by felixmosh** [comment]
It's kinda critical, since it breaks the app.
I've put a resolution to `v1.2.129` for now.

**Comment by ruessej** [comment]
This is fixed in `v1.2.135` see: https://play.swc.rs/?version=1.2.135&code=H4sIAAAAAAAAA0utKMgvKlFIzSvNVXBOzMkJqSxIVajmUlBQd%2FTxUVewVVAHCqrrgAR8I%2BOdgYLBYNHcSt1koEwxRMrPPyTe0S843DXI1QUsnZdfopuYV1yeWpSaAlRSywUA47fZDWgAAAA%3D&config=H4sIAAAAAAAAA0WNSwrDMAxE76J1ts0id%2BghjKsEF%2F%2FQKBBjfPfawSU7aebpqdIXlrZK2QhYxoQS1Vy0kZbMsOKy0kKKHu3Gg1tfjBysHWG8eudTAs92oeCi28sw2RSyMPBUJh7%2BT7YuCulzjqDez27hSu1xzDuH9wRVTm4%2FOkwQUrUAAAA%3D

**Comment by kdy1** [comment]
Thank you!
